src/admin.tsx

```diff
@@ -2,60 +2,153 @@ import React, { useContext, useEffect, useState } from 'react';
-import { collection, getFirestore, onSnapshot } from "firebase/firestore";
+import { collection, getFirestore, onSnapshot, doc, updateDoc } from "firebase/firestore";
 import AppBar from '@mui/material/AppBar';
 import Box from '@mui/material/Box';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import AddIcon from '@mui/icons-material/Add';
+import IconButton from '@mui/material/IconButton';
+import Tooltip from '@mui/material/Tooltip';
 import { TreeView } from '@mui/x-tree-view/TreeView';
 import { TreeItem } from '@mui/x-tree-view/TreeItem';
 import { Ambiance, AmbianceCategory } from './config/ambiance/types';
+import BuildCircleIcon from '@mui/icons-material/BuildCircle';
+import VideoEditor from './components/VideoEditor';
+import { AMBIANCE_COLLECTION } from './constants';
+import cloneDeep from 'lodash.clonedeep';
 
-const ambianceCollection: string = "ambiance";
+type SubcategoryMap = Record<string, Ambiance[]>;
+
+/**
+ * friendlyName is ambiance category name
+ */
+type SubcategoryGroupingType = {
+  friendlyName: string;
+  videosBySubcategory: SubcategoryMap;
+}
+
+/**
+ * Key is documentId
+ */
+type AmbianceDisplayType = Record<string, SubcategoryGroupingType>;
 
 export default function AdminPage() {
   const db = getFirestore();
   const { currentUser, signOut } = useContext(AuthContext);
-  const [data, setData] = useState<AmbianceCategory[]>();
+  const [data, setData] = useState<Record<string, AmbianceCategory>>();
 
   useEffect(() => {
-    const unsubscribe = onSnapshot(collection(db, ambianceCollection), (snapshot) => {
-      setData((snapshot.docs.map(doc => doc.data() as AmbianceCategory)));
+    const unsubscribe = onSnapshot(collection(db, AMBIANCE_COLLECTION), (snapshot) => {
+      const dataMap = snapshot.docs.reduce<Record<string, AmbianceCategory>>((obj, item) => {
+        obj[item.id] = item.data() as AmbianceCategory
+
+        return obj;
+      }, {});
+
+      setData(dataMap);
     });
 
     return unsubscribe;
-  }, [db]);
+  }, [db, setData]);
+
+  
+  const editVideo = async (documentId: string, videoUrlCode: string, newData: Ambiance) => {
+    if (!data) {
+      return;
+    }
+
+    const newVideos = cloneDeep(data[documentId].videos);
+    const videoIndex = newVideos.findIndex(video => video.code === videoUrlCode);
+    newVideos[videoIndex] = newData;
+
+    const docRef = doc(db, AMBIANCE_COLLECTION, documentId);
+    await updateDoc(docRef, {
+      videos: newVideos
+    });
+ 
+  }
+
+  const deleteVideo = async (documentId: string, videoUrlCode: string) => {
+    if (!data) {
+      return;
+    }
+
+    const newVideos = cloneDeep(data[documentId].videos)
+      .filter(video => video.code !== videoUrlCode);
+
+    const docRef = doc(db, AMBIANCE_COLLECTION, documentId);
+    await updateDoc(docRef, {
```

Comment:
Since this is the last thing we do, instead of awaiting here we can just return the result of this and allow the user of this function to `await` if they want to. If they dont want to, this will increase performance